### PR TITLE
constraints, create table: improve clarity and update out of date info

### PR DESCRIPTION
**Pull request by TomShawn**
### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v4.0 (TiDB 4.0 versions)
- [x] v3.1 (TiDB 3.1 versions)
- [x] v3.0 (TiDB 3.0 versions)
- [x] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs/pull/3582
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Have version specific changes 
- [ ] Might cause conflicts

### CODE REVIEW

File: sql-statements/sql-statement-create-table.md [hunk 1]
```diff
@@ -70,257 +70,120 @@ aliases: ['/docs-cn/dev/sql-statements/sql-statement-create-table/','/docs-cn/de
...
-|`CHARACTER SET` |指定该表所使用的字符集                | `CHARACTER SET` = 'utf8mb4'|
+|`CHARACTER SET` |指定该表所使用的[字符集](/character-set-and-collation.md)                | `CHARACTER SET` = 'utf8mb4'|
 |`COLLATE`       |指定该表所使用的字符集排序规则        | `COLLATE` = 'utf8mb4_bin'|
 |`COMMENT`       |注释信息                              | `COMMENT` = 'comment info'|
 
 > **注意：**
 >
-> 在 TiDB 配置文件中，`split-table` 默认开启。当该配置项开启时，建表操作会为每个表建立单独的 Region，详情参见 [TiDB 配置文件描述](/tidb-configuration-file.md) 。
+> + 在 TiDB 配置文件中，`split-table` 默认开启。当该配置项开启时，建表操作会为每个表建立单独的 Region，详情参见 [TiDB 配置文件描述](/tidb-configuration-file.md)。
+> + TiDB 会解析并忽略其他 `table_option` 参数，例如 `AVG_ROW_LENGTH`、`CHECKSUM`、`COMPRESSION`、`CONNECTION`、`DELAY_KEY_WRITE`、`ENGINE`、`KEY_BLOCK_SIZE`、`MAX_ROWS`、`MIN_ROWS`、`ROW_FORMAT` 和 `STATS_PERSISTENT`。
```
Comment (yikeke) [inline sql-statements/sql-statement-create-table.md:89]: 这句话还是建议和[英文放的位置](https://github.com/pingcap/docs/pull/3582/files#diff-bbf284717dcd1f4f14ff650bf8064552R73)保持一致吧，一是方便维护，二是保持逻辑的连贯性。

File: sql-statements/sql-statement-create-table.md [hunk 2]
```diff
@@ -70,257 +70,120 @@ aliases: ['/docs-cn/dev/sql-statements/sql-statement-create-table/','/docs-cn/de
...
 
 ## MySQL 兼容性
 
 * TiDB 不支持临时表，对于 `CREATE TEMPORARY TABLE` 语法，会忽略 `TEMPORARY` 关键字。
 * 支持除空间类型以外的所有数据类型。
 * 不支持 `FULLTEXT`，`HASH` 和 `SPATIAL` 索引。
-* `KEY_BLOCK_SIZE` 和 `ENGINE` 属性可被解析，但会被忽略。
-* `index_col_name` 属性支持 length 选项，最大长度默认限制为 3072 字节。此长度限制可以通过配置项 `max-index-length` 更改，具体请参阅 [TiDB 配置文件描述](/tidb-configuration-file.md#max-index-length)。
-* `index_col_name` 属性支持 `ASC` 和 `DESC` 的索引排序选项。
+* `index_col_name` 属性支持 length 选项，以与 MySQL 兼容。最大长度默认限制为 3072 字节。此长度限制可以通过配置项 `max-index-length` 更改，具体请参阅 [TiDB 配置文件描述](/tidb-configuration-file.md#max-index-length)。
```
Comment (yikeke) [inline sql-statements/sql-statement-create-table.md:176]:
```suggestion
* 为了与 MySQL 兼容，`index_col_name` 属性支持 length 选项，最大长度默认限制为 3072 字节。此长度限制可以通过配置项 `max-index-length` 更改，具体请参阅 [TiDB 配置文件描述](/tidb-configuration-file.md#max-index-length)。
```

File: sql-statements/sql-statement-create-table.md [hunk 3]
```diff
@@ -70,257 +70,120 @@ aliases: ['/docs-cn/dev/sql-statements/sql-statement-create-table/','/docs-cn/de
...
+* `index_col_name` 属性支持 `ASC` 和 `DESC` 的索引排序选项。TiDB 会解析但忽略这两个选项。
```
Comment (yikeke) [inline sql-statements/sql-statement-create-table.md:177]:
```suggestion
* 为了与 MySQL 兼容，TiDB 会解析但忽略 `index_col_name` 属性的 `[ASC | DESC]` 索引排序选项。
```

**Review by yikeke — COMMENTED** [review]
Rest LGTM

File: sql-statements/sql-statement-create-table.md [hunk 4]
```diff
@@ -70,257 +70,120 @@ aliases: ['/docs-cn/dev/sql-statements/sql-statement-create-table/','/docs-cn/de
...
-  | ENGINE [=] engine_name
-  | KEY_BLOCK_SIZE [=] value
-  | MAX_ROWS [=] value
-  | MIN_ROWS [=] value
-  | ROW_FORMAT [=] {DEFAULT|DYNAMIC|FIXED|COMPRESSED|REDUNDANT|COMPACT}
-  | STATS_PERSISTENT [=] {DEFAULT|0|1}
-```
-
-`table_option` 目前支持的只有 `AUTO_INCREMENT`、`SHARD_ROW_ID_BITS`（详情介绍请参考 [SHARD_ROW_ID_BITS](/shard-row-id-bits.md)）、`PRE_SPLIT_REGIONS`、`CHARACTER SET`、`COLLATE` 和 `COMMENT`，其它只是语法上支持。具体内容参考下表，各个子句之间用逗号隔开。
+TiDB 支持以下 `table_option`：
```
Comment (TomShawn) [inline sql-statements/sql-statement-create-table.md:73]:
```suggestion
TiDB 支持以下 `table_option`。TiDB 会解析并忽略其他 `table_option` 参数，例如 `AVG_ROW_LENGTH`、`CHECKSUM`、`COMPRESSION`、`CONNECTION`、`DELAY_KEY_WRITE`、`ENGINE`、`KEY_BLOCK_SIZE`、`MAX_ROWS`、`MIN_ROWS`、`ROW_FORMAT` 和 `STATS_PERSISTENT`。
```

File: sql-statements/sql-statement-create-table.md (hunk 1 above)
Comment (TomShawn) [inline sql-statements/sql-statement-create-table.md:89]:
```suggestion
```

File: sql-statements/sql-statement-create-table.md [hunk 5]
```diff
@@ -70,257 +70,119 @@ aliases: ['/docs-cn/dev/sql-statements/sql-statement-create-table/','/docs-cn/de
...
 |`AUTO_RANDOM_BASE`|用来指定 AutoRandom 自增部分的初始值，该参数可以被认为属于内部接口的一部分，对于用户而言请忽略 |`AUTO_RANDOM_BASE` = 0|
-|`CHARACTER SET` |指定该表所使用的字符集                | `CHARACTER SET` = 'utf8mb4'|
+|`CHARACTER SET` |指定该表所使用的[字符集](/character-set-and-collation.md)                | `CHARACTER SET` = 'utf8mb4'|
 |`COLLATE`       |指定该表所使用的字符集排序规则        | `COLLATE` = 'utf8mb4_bin'|
 |`COMMENT`       |注释信息                              | `COMMENT` = 'comment info'|
 
 > **注意：**
 >
-> 在 TiDB 配置文件中，`split-table` 默认开启。当该配置项开启时，建表操作会为每个表建立单独的 Region，详情参见 [TiDB 配置文件描述](/tidb-configuration-file.md) 。
+> + 在 TiDB 配置文件中，`split-table` 默认开启。当该配置项开启时，建表操作会为每个表建立单独的 Region，详情参见 [TiDB 配置文件描述](/tidb-configuration-file.md)。
```
Comment (TomShawn) [inline sql-statements/sql-statement-create-table.md:88]:
```suggestion
> 在 TiDB 配置文件中，`split-table` 默认开启。当该配置项开启时，建表操作会为每个表建立单独的 Region，详情参见 [TiDB 配置文件描述](/tidb-configuration-file.md)。
```

**Review by yikeke — APPROVED** [review]
LGTM

**Review by AilinKid — COMMENTED** [review]
LGTM

**Pull request merged**